charmcraft/commands/store/__init__.py

```diff
@@ -240,3 +240,49 @@ def run(self, parsed_args):
         table = tabulate(data, headers=headers, tablefmt='plain', numalign='left')
         for line in table.splitlines():
             logger.info(line)
+
+
+class ReleaseCommand(BaseCommand):
+    """Release a charm revision to specific channels."""
+    name = 'release'
+    help_msg = "relase a charm revision to one or more channels"
+
+    def fill_parser(self, parser):
+        """Add own parameters to the general parser."""
+        parser.add_argument(
+            'channels', metavar='channel', nargs='+',
+            help="the channel(s) to release to")
+        parser.add_argument('--name', help="the name of the charm to get revisions")
+        parser.add_argument(
+            '--revision', type=int,
+            help="the revision to release (defaults to latest)")
+
+    def run(self, parsed_args):
+        """Run the command."""
+        store = Store()
+
+        if parsed_args.name:
+            charm_name = parsed_args.name
+        else:
+            charm_name = get_name_from_metadata()
+            if charm_name is None:
+                raise CommandError(
+                    "Can't access name in 'metadata.yaml' file. The 'release' command needs to "
+                    "be executed in a valid project's directory, or indicate the charm name with "
+                    "the --name option.")
+
+        if parsed_args.revision:
+            revision = parsed_args.revision
+        else:
+            # find out which is the latest revision for the charm
+            revisions = store.list_revisions(charm_name)
```

Comment:
Over time, the list of all revisions grows without bound. Is there a way to have the store tell us the latest revisions rather than telling us all of them?
I'm concerned when you've released 1000 versions of a charm that it takes a while and generates a lot of data we don't care about.

Reply:
Processing a list of a 1000 items should be fast. Currently there are no ways of requesting that to the store, but they may grow something for this, as the processing time for them would be worse.

IOW, I'm more concerned on how much time the Store endpoint will take to answer, than in the time we take to process the response...

Reply:
it's not about where the slowness is, but that it's slow for the user. Let's ask the store, maybe they agree and make it a priority (hopefully its not a massive amount of work)

Reply:
Will do